From 1a527660dd6a127711ba53d685a78448940c2593 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Wed, 10 Sep 2025 16:54:06 +0200
Subject: [PATCH] kcm/updates: don't ask when to apply the updates on KDE Linux

Much like in Kinoite, KDE Linux's updates always happen upon reboot. Do
not add an inactionable option.

BUG: 508556
---
 kcm/CMakeLists.txt |  6 ++++++
 kcm/ui/main.qml    |  2 +-
 kcm/updates.cpp    |  4 ++--
 kcm/updates.h      | 10 +++++-----
 4 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/kcm/CMakeLists.txt b/kcm/CMakeLists.txt
index 0fe188586..745f8ceb8 100644
--- a/kcm/CMakeLists.txt
+++ b/kcm/CMakeLists.txt
@@ -25,3 +25,9 @@ target_link_libraries(kcm_updates PRIVATE
     KF6::ConfigGui
     KF6::KCMUtilsQuick
 )
+
+if(BUILD_RpmOstreeBackend OR BUILD_SystemdSysupdateBackend)
+    target_compile_definitions(kcm_updates PRIVATE UPDATE_NEEDS_REBOOT=1)
+else()
+    target_compile_definitions(kcm_updates PRIVATE UPDATE_NEEDS_REBOOT=0)
+endif()
diff --git a/kcm/ui/main.qml b/kcm/ui/main.qml
index 4defaece0..ff64bf1ba 100644
--- a/kcm/ui/main.qml
+++ b/kcm/ui/main.qml
@@ -118,7 +118,7 @@ SimpleKCM {
             spacing: 0
             Kirigami.FormData.label: i18n("Apply system updates:")
             Kirigami.FormData.buddyFor: offlineUpdatesOption
-            visible: !kcm.isRpmOstree
+            visible: !kcm.mandatoryRebootAfterUpdate
             enabled: !kcm.discoverSettings.isUseOfflineUpdatesImmutable
 
             QQC2.RadioButton {
diff --git a/kcm/updates.cpp b/kcm/updates.cpp
index 0b2bb5e4d..a269ca3b9 100644
--- a/kcm/updates.cpp
+++ b/kcm/updates.cpp
@@ -38,9 +38,9 @@ DiscoverSettings *Updates::discoverSettings() const
     return m_discoverData->settings();
 }
 
-bool Updates::isRpmOstree() const
+bool Updates::mandatoryRebootAfterUpdate() const
 {
-    return QFile::exists(QStringLiteral("/run/ostree-booted"));
+    return UPDATE_NEEDS_REBOOT;
 }
 
 #include "moc_updates.cpp"
diff --git a/kcm/updates.h b/kcm/updates.h
index adc786e24..aeb68254d 100644
--- a/kcm/updates.h
+++ b/kcm/updates.h
@@ -20,7 +20,7 @@ class Updates : public KQuickManagedConfigModule
     Q_OBJECT
     Q_PROPERTY(UpdatesSettings *updatesSettings READ updatesSettings CONSTANT)
     Q_PROPERTY(DiscoverSettings *discoverSettings READ discoverSettings CONSTANT)
-    Q_PROPERTY(bool isRpmOstree READ isRpmOstree CONSTANT)
+    Q_PROPERTY(bool mandatoryRebootAfterUpdate READ mandatoryRebootAfterUpdate CONSTANT)
 
 public:
     explicit Updates(QObject *parent, const KPluginMetaData &data);
@@ -28,10 +28,10 @@ public:
     UpdatesSettings *updatesSettings() const;
     DiscoverSettings *discoverSettings() const;
 
-    /* Returns true if we're running on an rpm-ostree managed systems. Used to
-     * show/hide PackageKit specific settings and show/hide rpm-ostree specific
-     * settings only on systems where those are relevant.*/
-    bool isRpmOstree() const;
+    /* Returns true if we're running on a system that needs rebooting after updating.
+     * It is used to hide the setting asking the user when to apply the updates
+     * because it's mandatory at times.*/
+    bool mandatoryRebootAfterUpdate() const;
 
 private:
     UpdatesData *const m_data;
-- 
GitLab

